src/site/content/en/blog/one-line-layouts/index.md

```diff
@@ -0,0 +1,319 @@
+
+```css/1
+.parent {
+  width: clamp(23ch, 50%, 46ch);
+}
+```
+
+The minimum size here is `23ch` or 23 character units, and the maximum size is `46ch`, 46 characters. [Character width units](https://meyerweb.com/eric/thoughts/2018/06/28/what-is-the-css-ch-unit/) are based on the font size of the element (specifically the with of the '0' glyph). The 'actual' size is 50%, which represents 50% of this element's parent width.
+
+What the `clamp()` function is doing here is enabling this element to retain a 50% width *until* 50% is either greater than `46ch` (on wider viewports), or smaller than `23ch` (on smaller viewports). You can see that as I stretch and shrink the parent size, the width of this card increases to its clamped maximum point and decreases to its clamped minimum. It then stays centered in the parent since we've applied additional properties to center it. This enables more legible layouts, as the text won't be too wide (above `46ch`) or too squished and narrow (less than `23ch`).
+
+This is also a great way to implement responsive typography. For example, you could write: `font-size: clamp(1.5rem, 20vw, 3rem)`. In this case, the font-size of a headline would always stay clamped between 1.5rem and 3rem but would grow and shrink based on the `20vw` actual value to fit the width of of the viewport.
+
+This is a great technique to ensure legibility with a minimum and maximum size value, but remember it is not supported in all modern browsers so make sure you have fallbacks and do your testing.
+
+## 10. Respect for Aspect: `aspect-ratio: <width> / <height>`
+
+<figure class='w-figure'>
+<video controls autoplay loop muted class='w-screenshot'>
+    <source src='https://storage.googleapis.com/web-dev-assets/one-line-layouts/10-aspectratio.mp4'>
+  </video>
+</figure>
+
+And finally, this last layout tool is the most experimental of the bunch. It was recently introduced to Chrome Canary in Chromium 84+, and there is active effort from Firefox in getting this implemented, but it is not currently in any stable browser editions.
+
+I do want to mention this, though, because it is such a frequently met problem. And that is just simply maintaining the aspect ratio of an image. 
+
+With the aspect-ratio property, as I resize the card, the green visual block maintains this 16 x 9 aspect ratio. We are Respecting the Aspect Ratio with `aspect-ratio: 16 / 9`. 
+
+```css/1
+.video {
+  aspect-ratio: 16 / 9;
+}
+```
+
+To maintain a 16 x 9 aspect ratio without this property, you'd need to use a [padding-top hack](https://css-tricks.com/aspect-ratio-boxes/) and give it a padding of `56.25%` to set a top-to-width ratio. We will soon have a property for this to avoid the hack and the need to calculate the percentage. You can make a square with `1 / 1` ratio, a 2 to 1 ratio with `2 / 1`, and really just anything you need for this image to scale with a set size ratio.
```

Comment:
```suggestion
To maintain a 16 x 9 aspect ratio without this property, you'd need to use a [`padding-top` hack](https://css-tricks.com/aspect-ratio-boxes/) and give it a padding of `56.25%` to set a top-to-width ratio. We will soon have a property for this to avoid the hack and the need to calculate the percentage. You can make a square with `1 / 1` ratio, a 2 to 1 ratio with `2 / 1`, and really just anything you need for this image to scale with a set size ratio.
```